test/val/val_id_test.cpp

```diff
@@ -2531,6 +2531,327 @@ OpFunctionEnd)";
                         "as an operand of <id> '24'."));
 }
 
+// Valid: Get a float in a matrix using CompositeExtract.
+// Valid: Insert float into a matrix using CompositeInsert.
+TEST_F(ValidateIdWithMessage, CompositeExtractInsertGood) {
+  ostringstream spirv;
+  spirv << kGLSL450MemoryModel << opAccessChainSpirvSetup << std::endl;
```

Comment:
opAccessChainSpirvSetup   should only be for AccessChain  tests.  ?

How about a rename to kDeeplyNestedStructureSetup.
Also, as per style guide, you should not have static duration non-POD values.  Make it a char[] instead.